cmd/modern/root/start.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package root
+
+import (
+	"github.com/microsoft/go-sqlcmd/internal/cmdparser"
+	"github.com/microsoft/go-sqlcmd/internal/config"
+	"github.com/microsoft/go-sqlcmd/internal/container"
+)
+
+type Start struct {
+	cmdparser.Cmd
+
+	errorLogEntryToWaitFor string
+}
+
+func (c *Start) DefineCommand(...cmdparser.CommandOptions) {
+	options := cmdparser.CommandOptions{
+		Use:   "start",
+		Short: "Start current context",
+		Examples: []cmdparser.ExampleOptions{
+			{
+				Description: "Start the current context",
+				Steps:       []string{`sqlcmd start`}},
+		},
+		Run: c.run,
+	}
+
+	c.Cmd.DefineCommand(options)
+	c.AddFlag(cmdparser.FlagOptions{
+		String:        &c.errorLogEntryToWaitFor,
+		DefaultString: "Recovery is complete",
+		Name:          "errorlog-wait-line",
+		Usage:         "Line in errorlog to wait for before exiting this process",
```

Comment:
this error log reference seems very SQL Server specific.  Why would we expect the user to provide this, when each implementation of "start" should know what its target's behavior is?  And the error log might be in another language.

Reply:
Yes, and it doesn't actually work either, which is very strange.  i.e. even if we wait here until this entry in the error log, there is still a few seconds before a connect will succeed.

So we can remove this completely from here, because it isn't working.  But I am using the same technique in the `sqlcmd create` section where I might be able to rely on it being US english on first start.  

But I agree with the sentiment, this is hack, because I don't know of a way to actually know SQL is  ready for connections.  

From memory in our test frameworks we wrote a tool called Wait4Sql.exe that listened for a named share memory section? that meant connections could now succeed.

Should I clone that Wait4Sql technique here.  Not sure?

Reply:
I'd not block at all, by default, and just make Start completely asynchronous. The script author might want to run a bunch of non-SQL commands while SQL is starting up. The open/connect call should be what blocks, for however long a timeout the user has specified.

Reply:
Ok, for now I'll remove all this code that is trying to wait until SQL is ready.

Because we can't control all the clients that are going to connect after the start command, I have a feeling this won't be a  good experience.  But we can tackle that later maybe with a Wait4Sql.exe type approach